AdobeStockAsset/Model/SaveAsset.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Model;
+
+use Magento\AdobeStockAssetApi\Api\AssetRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\CategoryRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\CreatorRepositoryInterface;
+use Magento\AdobeStockAssetApi\Api\Data\AssetInterface;
+use Magento\Framework\Exception\AlreadyExistsException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class SaveAsset
+ *
+ * Service for saving asset
+ */
+class SaveAsset
+{
+    /**
+     * @var AssetRepositoryInterface
+     */
+    private $assetRepository;
+
+    /**
+     * @var CategoryRepositoryInterface
+     */
+    private $categoryRepository;
+
+    /**
+     * @var CreatorRepositoryInterface
+     */
+    private $creatorRepository;
+
+    /**
+     * SaveAsset constructor.
+     *
+     * @param AssetRepositoryInterface $assetRepository
+     * @param CreatorRepositoryInterface $creatorRepository
+     * @param CategoryRepositoryInterface $categoryRepository
+     */
+    public function __construct(
+        AssetRepositoryInterface $assetRepository,
+        CreatorRepositoryInterface $creatorRepository,
+        CategoryRepositoryInterface $categoryRepository
+    ) {
+        $this->assetRepository = $assetRepository;
+        $this->creatorRepository = $creatorRepository;
+        $this->categoryRepository = $categoryRepository;
+    }
+
+    /**
+     * Save asset
+     *
+     * @param AssetInterface $asset
+     * @throws AlreadyExistsException
+     */
+    public function execute(AssetInterface $asset): void
+    {
+        $category = $asset->getCategory();
+        if ($category !== null && !$this->isCategorySaved($category->getId())) {
```

Comment:
isSaved checks can be skipped for all entities to speed up the processing (current resource models are using `INSERT IGNORE` operation)